src/main/java/com/ericsson/ei/controller/model/ParseInstanceInfoEI.java

```diff
@@ -137,16 +139,41 @@ public void init() throws IOException {
     @Component
     private class MongoDbValues {
         @Getter
-        @Value("${spring.data.mongodb.host}")
-        private String host;
-
-        @Getter
-        @Value("${spring.data.mongodb.port}")
-        private String port;
+        @Value("${spring.data.mongodb.uri}")
+        private String uri;
 
         @Getter
         @Value("${spring.data.mongodb.database}")
         private String database;
+
+        @PostConstruct
+        public void init() throws IOException {
+            if (uri != null) {
+                String secret = extractSecretFromUri(uri);
+                String modifiedUri = uri.replaceFirst(secret, "hidden_secret");
+                uri = modifiedUri;
+            }
```

Comment:
Maybe lay this is a method called "obscureUriForUser" or something better to make it clear what this does.

Reply:
Moved to its own class, although name is not osbucre

Reply:
Moved to own class.